sympy/vector/vector.py

```diff
@@ -418,19 +418,31 @@ def _div_helper(one, other):
 
 def get_postprocessor(cls):
     def _postprocessor(expr):
-        vec_class = {Add: VectorAdd}[cls]
+        vec_class = {Add: VectorAdd, Mul: VectorMul}[cls]
         vectors = []
+        nonvectors = []
         for term in expr.args:
             if isinstance(term.kind, VectorKind):
-                vectors.append(term)
+                if isinstance(term, (BaseVector, VectorMul, VectorAdd, VectorZero)):
+                    vectors.append(term)
+                else:
+                    nonvectors.append(term)
+            else:
+                nonvectors.append(term)
+
+        if not vectors:
+            return cls._from_args(expr.args)
 
         if vec_class == VectorAdd:
             return VectorAdd(*vectors).doit(deep=False)
+        if vec_class == VectorMul:
+            return VectorMul(*nonvectors, *vectors).doit(deep=False)
```

Comment:
I don't see why this needs to separate vectors and nonvectors at all if they are all going to be passed to VectorMul.

Think more carefully about how to write this code because I think it can be much simpler than this.